_source/_posts/2019-04-26-tutorial-build-universal-apps-with-nuxt.md

```diff
@@ -499,44 +486,91 @@ require('dotenv').config()
 
 ```bash
-npm i nuxt-oauth
+npm i @nuxtjs/auth-next
+npm install @nuxtjs/axios
 ```
 
-> Note: there is an official [Nuxt Community Auth module](https://auth.nuxtjs.org/) which supports OAuth. However, as of this writing, it does not include a nonce on requests for tokens. Therefore, it is not standards compliant.
-
-Once that is installed, you need to configure its behavior in the `nuxt.config.js`. First, add `nuxt-oauth`  to the array of modules.
+Once they are installed, you need to configure their behavior in the `nuxt.config.js`. First, add both to the array of modules.
 
 ```javascript
-  modules: [
-    ['nuxt-oauth']
+    modules: [
+    '@nuxtjs/axios',
+    '@nuxtjs/auth-next'
   ],
 ```
 
-Then add a new `oauth` section to configure the OAuth library.
+Then add a new `auth` section to configure the OAuth library.
 
 ```javascript
-  oauth: {
-    sessionName: 'WidgetCoSession',
-    secretKey: process.env.SECRET_KEY,
-    oauthHost: process.env.OAUTH_ISSUER,
-    oauthClientID: process.env.CLIENT_ID,
-    oauthClientSecret: process.env.CLIENT_SECRET,
-    scopes: ['openid', 'profile'],
+  auth:{
+    strategies:{
+      okta:{
+        scheme: 'oauth2',
+        endpoints: {
+          authorization: process.env.OAUTH_ISSUER+"/v1/authorize",
+          token: process.env.OAUTH_ISSUER+"/v1/token",
+          userInfo: process.env.OAUTH_ISSUER+"/v1/userinfo",
+          logout: process.env.OAUTH_ISSUER+"/v1/logout"
+        },
+        token: {
+          property: 'access_token',
+          type: 'Bearer',
+          maxAge: 1800,
+        },
+        responseType: 'code',
+        grantType: 'authorization_code',
+        clientId: process.env.CLIENT_ID,
+        scope: ['openid', 'profile', 'email'],
+        codeChallengeMethod: 'S256',
+        autoLogout: true,
+        }
+    }
   },
 ```
 
-This implementation requires a Vuex store. Nuxt will not create the store by default, you must create an empty `index.vue` file in the `store` folder. Once Nuxt detects this file it will include the correct dependencies for you.
+This implementation requires a Vuex store. Nuxt will not create the store by default, you must create an empty `index.js` file in the `store` folder. Once Nuxt detects this file it will include the correct dependencies for you.
+
+The Nuxt/Auth middleware allows you to have multiple strategies configured at once. So you can support local login as well as integrating with OAuth identity providers like Okta. This
+means that you need to tell the middleware how to authenticate the user. To do this create a new page of `login.vue` in the `pages` folder with the following code.
+
+```javascript
+<template>
+  <div>
+    Redirecting you to Okta to login
+  </div>
+</template>
+
+<script>
+export default {
+    mounted: function(){
+      if(!this.$auth.loggedIn){
+        this.$auth.loginWith('okta')
+      }
+      else {
+        this.$router.push('/')
+      }
+    }
```

Comment:
Should this code block be outdented 2 spaces?

Reply:
Correct, addressed by a37adfe